ClosedXML/Excel/Ranges/XLRangeBase.cs

```diff
@@ -635,15 +635,19 @@ public XLCell FirstCellUsed(Boolean includeFormats, Func<IXLCell, Boolean> predi
                         if (!cell.IsEmpty(true)) return cell;
                     }
                 }
+            }
 
-                var intersectedRanges = Worksheet.MergedRanges.GetIntersectedRanges(RangeAddress).ToList();
-                if (intersectedRanges.Any())
-                {
-                    Int32 minRo = intersectedRanges.Min(r => r.RangeAddress.FirstAddress.RowNumber);
-                    Int32 minCo = intersectedRanges.Min(r => r.RangeAddress.FirstAddress.ColumnNumber);
+            var intersectedRanges = Worksheet.MergedRanges.GetIntersectedRanges(RangeAddress)
+                .Where(r => predicate == null || predicate(r.FirstCell())).ToList();
+            if (intersectedRanges.Any())
+            {
+                Int32 minRo = intersectedRanges.Min(r => r.RangeAddress.FirstAddress.RowNumber);
+                Int32 minCo = intersectedRanges.Min(r => r.RangeAddress.FirstAddress.ColumnNumber);
 
-                    return Worksheet.Cell(minRo, minCo);
-                }
+                if (0 < sp.Row && sp.Row < minRo) minRo = sp.Row;
```

Comment:
There's a nice extension method to simplify this: `if (sp.Row.Between(1, minRo - 1)) ...`

Reply:
Cool, thanks!